rust/arrow/src/datatypes.rs

```diff
@@ -156,6 +157,107 @@ make_type!(UInt64Type, u64, DataType::UInt64, 64, 0u64);
 make_type!(Float32Type, f32, DataType::Float32, 32, 0.0f32);
 make_type!(Float64Type, f64, DataType::Float64, 64, 0.0f64);
 
+macro_rules! make_temporal_type {
```

Comment:
Instead of this macro, can we reuse `make_type` - we'll need to move the 
```rust
impl ArrowNativeType for $native_ty {}
```
out though, but I think it's better than having a separate but almost the same macro.